.github/workflows/test.yml

```diff
@@ -1,61 +1,88 @@
-name: Cargo NEAR Checks And Tests
+name: CI
 
 on:
   push:
     branches: [ main ]
   pull_request:
 
 jobs:
-  test:
+  tests:
     runs-on: ${{ matrix.platform }}
-    name: "${{ matrix.platform }} ${{ matrix.toolchain }}"
     strategy:
       matrix:
         platform: [ubuntu-latest, macos-latest]
         toolchain: [stable, 1.56.0]
+
     steps:
       - name: Checkout Sources
         uses: actions/checkout@v2
+
       - name: "Install ${{ matrix.toolchain }} toolchain"
         uses: actions-rs/toolchain@v1
         with:
           profile: minimal
           toolchain: ${{ matrix.toolchain }}
+
       - name: Run tests
-        run: cargo test --verbose
+        run: cargo test --verbose --workspace --exclude cargo-near-integration-tests
+
+  integration-test:
+    name: "tests batch #${{ matrix.batch }} (${{ matrix.platform }}, ${{ matrix.toolchain }})"
+    runs-on: ${{ matrix.platform }}
+    strategy:
+      matrix:
+        platform: [ubuntu-latest, macos-latest]
+        toolchain: [stable, 1.56.0]
+        batch: [1, 2, 3, 4, 5]
 
-  lint:
-    name: Clippy and fmt
-    runs-on: ubuntu-latest
     steps:
       - name: Checkout Sources
         uses: actions/checkout@v2
-      - name: Install stable toolchain
+
+      - name: "Install ${{ matrix.toolchain }} toolchain"
         uses: actions-rs/toolchain@v1
-        with: 
+        with:
           profile: minimal
-          toolchain: stable
-          components: rustfmt, clippy
+          toolchain: ${{ matrix.toolchain }}
+
+      - name: Run tests
+        run: |
+          cargo test --package cargo-near-integration-tests -- --list --format terse \
+              | grep ': test$' | sed 's/\(.*\): .*$/\1/' \
+              > tests
+          share=$((`wc -l < tests` / 5))
+          units=$(tail -n +$(((share*(${{ matrix.batch }}-1))+1)) tests | head -n +$share | tr '\n' ' ')
+          cargo test --verbose --package cargo-near-integration-tests -- $units
+
+  fmt:
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Checkout Sources
+        uses: actions/checkout@v2
+
       - name: Check Formatting
         run: cargo fmt -- --check
+
+  clippy:
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Checkout Sources
+        uses: actions/checkout@v2
+
       - name: Check Clippy
         run: cargo clippy --tests -- -Dclippy::all
 
   audit:
-    name: Audit
     runs-on: ubuntu-latest
+
     steps:
       - name: Checkout Sources
         uses: actions/checkout@v2
-      - name: Install Toolchain
-        uses: actions-rs/toolchain@v1
-        with:
-          profile: minimal
-          toolchain: stable
-      - name: Install Audit
+
+      - name: Install Audit Tool
         run: cargo install cargo-audit
-      - name: Run Audit
-        uses: actions-rs/cargo@v1
-        with:
-          command: audit
-          args: --ignore RUSTSEC-2020-0071
+
+      - name: Run Audit Tool
+        run: cargo audit --ignore RUSTSEC-2020-0071
```

Comment:
note to self or for someone else to remember, if we remove the SDK dependency in https://github.com/near/cargo-near/pull/23, we can remove this ignore